pkgs/applications/editors/notepad-next/default.nix

```diff
@@ -0,0 +1,36 @@
+{ mkDerivation, lib, fetchFromGitHub, qmake, libsForQt5 }:
+
+mkDerivation rec {
+  pname = "notepad-next";
+  version = "0.5.1";
+
+  src = fetchFromGitHub {
+    owner = "dail8859";
+    repo = "NotepadNext";
+    rev = "v${version}";
+    sha256 = "sha256-J7Ngt6YtAAZsza2lN0d1lX3T8gNJHp60sCwwaLMGBHQ=";
+    fetchSubmodules = true;
```

Comment:
Would be even nicer if upstream optionally supported external packages instead of vendored dependencies. We already have lua packaged and it should not be that hard to package the rest.

Reply:
How can I verify if this is possible/is there a reference that I can suggest upstream how to do it? Maybe we can make an issue out of this in the end not blocking a merge?

Reply:
Yeah, this will be a bit more work upstream so issue should be fine. I am not really familiar with qmake but generally, there would be an option to enable using external dependencies ([example for CMake based project](https://github.com/NixOS/nixpkgs/pull/171934/files#diff-afc2a5b63cab6314c97002777c79efe92ad10559b2c482b95ecfff0aa574bf5eR161)), which when enabled, would find the dependencies using pkg-config instead of building them locally.

For example, here are Debian guidelines for projects concerning this: https://wiki.debian.org/UpstreamGuide#No_inclusion_of_third_party_code

Reply:
Thanks for the help, I reported it upstream with reference to your comment and referenced the issue in the code for future updates/changes 👍🏻  https://github.com/dail8859/NotepadNext/issues/135